sds_data_manager/lambda_code/SDSCode/dependency_config.py

```diff
@@ -628,173 +638,84 @@
         direction="DOWNSTREAM",
     ),
-    PreProcessingDependency(
-        primary_instrument="ultra",
-        primary_data_level="l1a",
-        primary_descriptor="90sensor-rates",
-        dependent_instrument="ultra",
-        dependent_data_level="l1b",
-        dependent_descriptor="90all",
-        relationship="HARD",
-        direction="DOWNSTREAM",
-    ),
-    # TODO: uncomment for post-SIT-3
-    # PreProcessingDependency(
-    #     primary_instrument="codice",
-    #     primary_data_level="l2",
-    #     primary_descriptor="todo",
-    #     dependent_instrument="ultra",
-    #     dependent_data_level="l1b",
-    #     dependent_descriptor="90all",
-    #     relationship="HARD",
-    #     direction="DOWNSTREAM",
-    # ),
-    # PreProcessingDependency(
-    #     primary_instrument="swapi",
-    #     primary_data_level="l2",
-    #     primary_descriptor="todo",
-    #     dependent_instrument="ultra",
-    #     dependent_data_level="l1b",
-    #     dependent_descriptor="90all",
-    #     relationship="HARD",
-    #     direction="DOWNSTREAM",
-    # ),
     PreProcessingDependency(
         primary_instrument="ultra",
         primary_data_level="l1b",
-        primary_descriptor="90sensor-de",
+        primary_descriptor="45sensor-extendedspin",
         dependent_instrument="ultra",
         dependent_data_level="l1c",
-        dependent_descriptor="90all",
+        dependent_descriptor="45sensor-pset",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
     PreProcessingDependency(
         primary_instrument="ultra",
         primary_data_level="l1b",
-        primary_descriptor="90sensor-extendedspin",
+        primary_descriptor="45sensor-cullingmask",
         dependent_instrument="ultra",
         dependent_data_level="l1c",
-        dependent_descriptor="90all",
+        dependent_descriptor="45sensor-pset",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
+    # L1a histogram kicking off l1c histogram is a special case here
     PreProcessingDependency(
         primary_instrument="ultra",
-        primary_data_level="l1b",
-        primary_descriptor="90sensor-cullingmask",
+        primary_data_level="l1a",
+        primary_descriptor="45sensor-histogram",
         dependent_instrument="ultra",
         dependent_data_level="l1c",
-        dependent_descriptor="90all",
+        dependent_descriptor="45sensor-histogram",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
     PreProcessingDependency(
         primary_instrument="ultra",
         primary_data_level="l1b",
-        primary_descriptor="90sensor-badtimes",
+        primary_descriptor="45sensor-cullingmask",
         dependent_instrument="ultra",
         dependent_data_level="l1c",
-        dependent_descriptor="90all",
+        dependent_descriptor="45sensor-histogram",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
+    # TODO: Add Ultra-90 products
```

Comment:
Might want to write a function to do this automatically for you.

Reply:
totally